src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -1008,6 +1375,47 @@ public OperationResult ProcessQueuedOperation(TOperation op)
                 return (wasCompleted ? OperationResult.Completed : OperationResult.Cancelled);
             }
 
+            public void ProcessBatchedOperation(TOperation op)
+            {
+                using (Lock())
+                {
+                    _batchedOperationsCount--;
+
+                    if (_state == QueueState.Stopped)
+                    {
+                        Debug.Assert(_tail == null);
+                        return;
+                    }
+
+                    if (op.IsWaiting)
+                    {
+                        _state = QueueState.Waiting;
+                    }
+                    else if (op == _tail)
+                    {
+                        // No more operations to process
+                        _tail = null;
+                        _state = QueueState.Ready;
+                        _sequenceNumber++;
+                    }
+                    else if (_tail.Next == op)
+                    {
+                        // Pop current operation and advance to next
+                        _tail.Next = op.Next;
+                    }
+                    else
+                    {
+                        throw new Exception($"Something went wrong, {_batchedOperationsCount}");
```

Comment:
I was never able to get to this point, but I am still not sure if it's possible or not. this requires more testing